### DISCUSSION TITLE
Wait for Cirrus CI job to complete and then start a GH workflow

### DISCUSSION THREAD

**Discussion by Maverick-I0** [community/community#27037] [Actions]
<p>I have tests that run on Cirrus CI, and then a workflow that builds my app inside GitHub repo.</p>
<p>I wanted to know if there is a way to wait till all the Cirrus CI and other workflows are completed. Then run the deployment workflow in github.</p>

**Comment by minecraftchest1** [comment]
<p>Using the github api, there is a way to trigger a workflow run.</p>

**Comment by ghost** [comment, marked as answer]
<p>I imagine you could use <code>repository_dispatch</code>:</p>
<p><a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#repository_dispatch" class="onebox" target="_blank" rel="noopener">https://docs.github.com/en/actions/reference/events-that-trigger-workflows#repository_dispatch</a></p>
<p>So once your tests are done on CircleCI, you can send a repository_dispatch to GitHub.</p>
<p>You also mentioned “wait till all the Cirrus CI and other workflows”. If those other workflows are GitHub Action workflows, you can also use <code>workflow_run</code>:</p>
<p>This event occurs when a workflow run is requested or completed, and allows you to execute a workflow based on the finished result of another workflow. A workflow run is triggered regardless of the result of the previous workflow.</p>
<p><a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#workflow_run" class="onebox" target="_blank" rel="noopener">https://docs.github.com/en/actions/reference/events-that-trigger-workflows#workflow_run</a></p>
